packages/next/build/babel/plugins/next-ssg-transform.ts

```diff
@@ -55,6 +59,17 @@ function decorateSsgExport(
   })
 }
 
+const checkIsSSG = (name: string, state: PluginState, toRemove: any) => {
+  if (ssgExports.has(name)) {
+    state.isPrerender = true
+
+    if (name === EXPORT_NAME_GET_SERVER_PROPS) {
+      state.isServerProps = true
```

Comment:
Being both `isPrerender` and `isServerProps` is weird here. 🤔

Reply:
Also, we should throw here at the loader level if both are defined. Better DX than waiting for post-build during require.

Reply:
Updated to only enable one or the other at a time and when one is already set to throw the error. Also added a test for this in `test/integration/mixed-ssg-serverprops-error`